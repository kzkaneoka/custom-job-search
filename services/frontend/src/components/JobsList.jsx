import React from "react";
import PropTypes from "prop-types";

const JobsList = props => {
  return (
    <div>
      <table className="table is-hoverable is-fullwidth">
        <thead>
          <tr>
            <th>Company</th>
            <th>Job Title</th>
            <th>Location</th>
            <th>Posted Date</th>
            <th>Link</th>
            <th />
          </tr>
        </thead>
        <tbody>
          {props.jobs.map(job => {
            return (
              <tr key={job.link}>
                <td>{job.company}</td>
                <td>{job.title}</td>
                <td>{job.location}</td>
                <td>{job.date}</td>
                <td>{job.link}</td>
              </tr>
            );
          })}
        </tbody>
      </table>
    </div>
  );
};

JobsList.propTypes = {
  jobs: PropTypes.array.isRequired
};

export default JobsList;
