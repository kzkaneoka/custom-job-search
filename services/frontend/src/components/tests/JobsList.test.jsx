import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import JobsList from "../JobsList";

afterEach(cleanup);

const jobs = [
  {
    company: "ABC",
    title: "Software Engineer",
    location: "San Francisco",
    date: "n/a",
    link: "https://www.indeed.com/abc"
  },
  {
    company: "XXX",
    title: "Backend Engineer",
    location: "San Jose",
    date: "n/a",
    link: "https://www.indeed.com/xxx"
  }
];

it("renders", () => {
  const { asFragment } = render(<JobsList jobs={jobs} />);
  expect(asFragment()).toMatchSnapshot();
});
