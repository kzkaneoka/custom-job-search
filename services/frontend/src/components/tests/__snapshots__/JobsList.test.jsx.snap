// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <table
      class="table is-hoverable is-fullwidth"
    >
      <thead>
        <tr>
          <th>
            Company
          </th>
          <th>
            Job Title
          </th>
          <th>
            Location
          </th>
          <th>
            Posted Date
          </th>
          <th>
            Link
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            ABC
          </td>
          <td>
            Software Engineer
          </td>
          <td>
            San Francisco
          </td>
          <td>
            n/a
          </td>
          <td>
            https://www.indeed.com/abc
          </td>
        </tr>
        <tr>
          <td>
            XXX
          </td>
          <td>
            Backend Engineer
          </td>
          <td>
            San Jose
          </td>
          <td>
            n/a
          </td>
          <td>
            https://www.indeed.com/xxx
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
